### PR TITLE
Add pokerbotic swar to zoo.

### PR DESCRIPTION
**Pull request by Scottbruceheart**
Copies over swar/metalog from pokerbotic, adds a shim test, add cmakelists entry for swar testing.

### CODE REVIEW

File: test/swar/BasicOperations.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+using namespace zoo;
+
+TEST_CASE(
+    "Basic SWAR",
+    "[swar]"
+) {
+    uint8_t v = 0x3;
+    SECTION("Simple Mask") {
+      auto mask = swar::makeBitmask<8, uint8_t>(v);
+      CHECK(mask == 0x08080808ull);
```
Comment (thecppzoo) [inline test/swar/BasicOperations.cpp:17]: In the interest of brevity, perhaps we should make a 32 bit or perhaps a 16 bit test instead, of nibbles?

File: inc/zoo/swar/SWAR.h [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+}
+
+static_assert(0x210 == popcount<0>(0x320), "");
+static_assert(0x4321 == popcount<1>(0xF754), "");
+static_assert(0x50004 == popcount<3>(0x3E001122), "");
+
+
+template<typename T> constexpr typename std::make_unsigned<T>::type msb(T v) {
+    return 8*sizeof(T) - 1 - __builtin_clzll(v);
+}
```
Comment (thecppzoo) [inline inc/zoo/swar/SWAR.h:111]: This should be called something like `bitIndex`, `msb` invites to think about the msb...

File: inc/zoo/swar/SWAR.h [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    constexpr auto msbMask = makeBitmask<Size, T>(msb);
+    constexpr auto subtraend = makeBitmask<Size, T>(N);
+    auto adjusted = v.value() | msbMask;
+    auto rv = adjusted - subtraend;
+    rv &= msbMask;
+    return rv;
+}
+
+static_assert(
+    0x80880008 == greaterEqualSWAR<3>(SWAR<4, uint32_t>(0x32451027)).value(),
```
Comment (thecppzoo) [inline inc/zoo/swar/SWAR.h:176]: These tests... let's improve this afterwards

File: test/swar/BasicOperations.cpp [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+TEST_CASE(
+    "Basic SWAR",
+    "[swar]"
+) {
+    uint8_t v = 0x8;
+    SECTION("Simple Mask") {
+      auto mask = swar::makeBitmask<8, uint32_t>(v);
+      CHECK(mask == 0x08080808ull);
+    }
+}
```
Comment (thecppzoo) [inline test/swar/BasicOperations.cpp:19]: Test the things taht can be tested at compilation time at compilation time.  As of now, there's no need to include catch here either

**Review by thecppzoo — APPROVED** [review]
I'd prefer you make tests at compilation time

File: inc/zoo/swar/SWAR.h (hunk 2 above)
Comment (Scottbruceheart) [reply inc/zoo/swar/SWAR.h:111]: I certainly thought msb meant 'most significant bit', msb_index does seem like a more clear name.

**Pull request merged**